### PR TITLE
Re-enable gzip compression for local development

### PR DESCRIPTION
**Pull request by umpox**
This PR re-enables compression on the local development server whilst still supporting streaming search.

**Context**
We had to disable gzip as it meant waiting for the search query to fully complete before the response was returned: https://github.com/sourcegraph/sourcegraph/pull/25404

**Changes**
- We now compress all resources locally (and get the performance benefits), except for the `/stream/search` endpoint. 
- The `web-standalone` variant commands now support streaming search. This was broken previously and was not updated as we maintain two separate DevServer configurations for different local development. We should aim to unify these.

### CODE REVIEW

**Review by eseliger — APPROVED** [review]
Nice, thanks for figuring this out!

File: client/web/dev/utils/should-compress-response.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import compression, { CompressionFilter } from 'compression'
+
+export const shouldCompressResponse: CompressionFilter = (request, response) => {
+    // Disable compression because gzip buffers the full response
+    // before sending it, which makes streaming search not stream.
+    if (request.path.startsWith('/search/stream')) {
+        return false
+    }
```
Comment (valerybugakov) [inline client/web/dev/utils/should-compress-response.ts:8]: 💪

**Review by valerybugakov — APPROVED** [review]
Thanks @umpox!

**Comment by camdencheek** [comment]
Awesome! Thanks for making this a real solution 😄

**Review by camdencheek — APPROVED** [review]

**Pull request merged**